sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -496,6 +496,14 @@ object IntervalUtils {
           state = TRIM_BEFORE_SIGN
         case TRIM_BEFORE_SIGN => trimToNextState(b, SIGN)
         case SIGN =>
+          currentValue = 0
+          fraction = 0
+          // We preset next state from SIGN to TRIM_BEFORE_VALUE. If we meet '.' in the SIGN state,
```

Comment:
The comment is useless from my point of view. It describes what is obvious from the code but does not say why.

Reply:
agree. @yaooqinn can you refine the comments to explain why? We should mention that it's to support `.1 second`

Reply:
OK